### PR TITLE
fix: allow usage of custom vid/cid

### PR DESCRIPTION
**Pull request by robertstypa**
Allow usage of custom vid cid

Ref: NCSDK-27373

Signed-off-by: Robert Stypa <robert.stypa@nordicsemi.no>

### CODE REVIEW

File: build_configuration/configuration.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    def _parse(self) -> None:
+        """Parse input .config file and populate the configuration dictionary."""
+        for config_line in self._config_data:
+            if re_result := self.config_value_pattern.match(config_line):
+                kconfig_name = re_result.group("kconfig_name")
+                kconfig_value = re_result.group("kconfig_value")
+                if kconfig_value == "y":
+                    # boolean value
+                    kconfig_value = True
+                elif kconfig_value.startswith("0x"):
```
Comment (tomchy) [inline build_configuration/configuration.py:39]: `and kconfig_value[2:].isdecimal()`?

File: build_configuration/configuration.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    # hexadecimal value
+                    kconfig_value = int(kconfig_value, base=16)
+                elif kconfig_value.startswith('"') and kconfig_value.endswith('"'):
+                    # string value
+                    kconfig_value = kconfig_value[1:-1]
+                elif kconfig_value.isdecimal():
+                    # int value
+                    kconfig_value = int(kconfig_value, base=10)
```
Comment (tomchy) [inline build_configuration/configuration.py:47]: Is it possible that `kconfig_value` or `kconfig_name` is `None`?
I'm wondering if we need to add a generic `else` here and raise an exception.

File: suit_generator/cmd_image.py [hunk 3]
```diff
@@ -183,6 +191,7 @@ class EnvelopeStorage:
         },
     ]
 
+    # fixme: how to handle user cid/vid
```
Comment (tomchy) [inline suit_generator/cmd_image.py:194]: Is it still valid?

File: build_configuration/configuration.py (hunk 1 above)
Comment (robertstypa) [reply build_configuration/configuration.py:39]: this part is for the hexadecimal values, `isdecimal()` does not cover A-F characters:
```
x = '0xFF'
x.startswith('0x')
True
x[2:].isdecimal()
False
```

File: build_configuration/configuration.py (hunk 2 above)
Comment (robertstypa) [reply build_configuration/configuration.py:47]: It seems that for `kconfig_name` and `kconfig_value` is not possible to have `None` values although for `konfig_value` there is a possibility to have an empty value or other values which won't be converted under `if_else` statement - in all mentioned cases `kconfig_value` should be stored as it is due to assignments in line 8: `kconfig_value = re_result.group("kconfig_value")', for example:
```
import re
def _test_code(config_line):
    print(f"{config_line=}")
    config_value_pattern = re.compile(r"(?P<kconfig_name>[A-Za-z0-9_]+)=(?P<kconfig_value>.*)")
    if re_result := config_value_pattern.match(config_line):
        kconfig_name = re_result.group("kconfig_name")
        kconfig_value = re_result.group("kconfig_value")
        if kconfig_value == "y":
            # boolean value
            kconfig_value = True
        elif kconfig_value.startswith("0x") and all(c in string.hexdigits for c in kconfig_value[2:]):
            # hexadecimal value
            kconfig_value = int(kconfig_value, base=16)
        elif kconfig_value.startswith('"') and kconfig_value.endswith('"'):
            # string value
            kconfig_value = kconfig_value[1:-1]
        elif kconfig_value.isdecimal():
            # int value
            kconfig_value = int(kconfig_value, base=10)

        print(f"{kconfig_name=}, {kconfig_value=}")
    else:
        print("No match")

_test_code("CONFIG_TEST=")
_test_code("CONFIG_TEST=\"TEST")
_test_code("CONFIG_TEST=TE\"ST")
_test_code("=")
_test_code("")
_test_code("not_valid_hex=0xWRRRRR")
```
```
config_line='CONFIG_TEST='
kconfig_name='CONFIG_TEST', kconfig_value=''
config_line='CONFIG_TEST="TEST'
kconfig_name='CONFIG_TEST', kconfig_value='"TEST'
config_line='CONFIG_TEST=TE"ST'
kconfig_name='CONFIG_TEST', kconfig_value='TE"ST'
config_line='='
No match
config_line=''
No match
config_line='not_valid_hex=0xWRRRRR'
kconfig_name='not_valid_hex', kconfig_value='0xWRRRRR'
```

File: suit_generator/cmd_image.py (hunk 3 above)
Comment (robertstypa) [reply suit_generator/cmd_image.py]: Removed!

**Review by tomchy — APPROVED** [review]

**Review by adsz-nordic — APPROVED** [review]

**Review by e-rk — APPROVED** [review]

**Review by kszromek-nordic — APPROVED** [review]

**Pull request merged**